### PR TITLE
*actually* support atom properly

### PR DESCRIPTION
**Pull request by ehzhang**
- add react-for-atom as a peer dependency
- don't check for window

cc @bolinfest

### CODE REVIEW

File: src/stubs/react/React.js [hunk 1]
```diff
@@ -12,7 +12,7 @@
 var React;
 
 // Check for Atom!
-if (typeof window === 'undefined' && typeof global === 'object' && global.atom){
+if (typeof global === 'object' && global.atom){
```
Comment (bolinfest) [inline src/stubs/react/React.js]: nit: space before `{`

**Comment by bolinfest** [comment]
LGTM. Running:

    !!(typeof global === 'object' && global.atom)

in Atom evaluates to `true`.

**Comment by cesarandreu** [comment]
Shouldn't `react-for-atom` be an [optional dependency](https://docs.npmjs.com/files/package.json#optionaldependencies)?

**Comment by ehzhang** [comment]
Thanks for the suggestion! Just added that.

**Comment by ehzhang** [comment]
cc @pieterv

**Comment by pieterv** [comment]
@bolinfest i'm not happy with the approach taken here, as a library special casing one particular consumers use case like isn't an maintainable solution for us given all the consumers we have and their needs. Its great we could unblock you for this but lets work together to find a way to alias the react module at your build level (babel has a way to do this) as we wont be able to maintain this special case long term and i would guess not all libraries are going to be so accommodating.

**Pull request merged**